### PR TITLE
Add script to generate Windows installers (.msi)

### PR DESCRIPTION
**Pull request by florisvdg**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,3 +26,6 @@ build:
 
 install:
 	go install ${BUILD_FLAGS} ./cmd/secrethub
+
+msi:
```
Comment (mackenbach) [inline Makefile]: Consider adding the `make msi` command to the goreleaser config as a pre or post build hook (not sure which would be preferred here, I think post). Then we can include the resulting `.msi` files in the release.

File: msi.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+NAME="SecretHub CLI"
+EXE=bin/secrethub.exe
+VERSION=$(git describe --tags $(git rev-list --tags --max-count=1))
+MANUFACTURER=SecretHub
+UPGRADE_CODE=7178CB18-B431-4A9D-B29E-D31E63A1CF77
```
Comment (mackenbach) [inline msi.sh]: Is this a static value?

File: msi.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+ICON="/dist/secrethub.png"
+
+curl -s -o dist/secrethub.png https://secrethub.io/img/secrethub-logo-rgb-shield-square.png
+mkdir -p dist/msi
+docker run -v $(pwd)/dist:/dist msi-packager /dist/windows_amd64 /dist/msi/secrethub_windows_amd64.msi -n $NAME -e $EXE -v $VERSION -m $MANUFACTURER -u $UPGRADE_CODE -i $ICON --local
+docker run -v $(pwd)/dist:/dist msi-packager /dist/windows_386 /dist/msi/secrethub_windows_386.msi -n $NAME -e $EXE -v $VERSION -m $MANUFACTURER -u $UPGRADE_CODE -i $ICON --local
```
Comment (mackenbach) [inline msi.sh]: Wow this looks nice! Love how simple and isolated it is with the docker build container.

File: Makefile (hunk 1 above)
Comment (florisvdg) [reply Makefile]: I tried that at first, but those hooks very limited. Can have another look later.

File: msi.sh (hunk 2 above)
Comment (florisvdg) [reply msi.sh]: Yes this is an arbitrary GUID to identify our app, so Windows knows it's same app across potential name changes.

**Comment by florisvdg** [comment]
It works, but it's not that pretty. It puts an icon on the desktop and it doesn't symlink the binary.

File: msi.sh (hunk 2 above)
Comment (mackenbach) [reply msi.sh]: Ah ok, got it 👍 Consider adding a comment as this is some weird arcane MSI knowledge.

**Comment by mackenbach** [comment]
Hmm, ok. Let's research how we can improve the installer. I think there are quite some options for setting e.g. the `PATH` value that the installer can do.

**Comment by jpcoenen** [comment]
This is coming very soon separately from this PR.

**Comment by jpcoenen** [comment]
An MSI installer for `v0.30.0` is now live at the [releases page](https://github.com/secrethub/secrethub-cli/releases/).

Direct download links:
[64 bit / amd64 / x86_x64](https://github.com/secrethub/secrethub-cli/releases/download/v0.30.0/secrethub-v0.30.0-windows-amd64.msi)
[32-bit / i386 / x86](https://github.com/secrethub/secrethub-cli/releases/download/v0.30.0/secrethub-v0.30.0-windows-386.msi)

**Pull request closed without merging**